src/main/java/org/rumbledb/runtime/functions/io/JsonDocFunctionIterator.java

```diff
@@ -74,10 +79,19 @@ public Item next() {
                     this.currentDynamicContextForLocalExecution.getRumbleRuntimeConfiguration(),
                     getMetadata()
                 );
-                JsonIterator object = JsonIterator.parse(is, 1024);
-                return ItemParser.getItemFromObject(object, getMetadata());
+                DslJson<Object> dslJson = new DslJson<Object>();
+                JsonReader<Object> object = dslJson.newReader();
+                object.process(is);
+                return ItemParser.getItemFromObject(object, getMetadata(), true);
             } catch (IteratorFlowException e) {
-                throw new IteratorFlowException(e.getJSONiqErrorMessage(), getMetadata());
+                RumbleException ex = new IteratorFlowException(e.getJSONiqErrorMessage(), getMetadata());
+                ex.initCause(e);
+            } catch (IOException e) {
+                RumbleException ex = new IteratorFlowException(
+                        "I/O Exception while binding the input stream",
+                        getMetadata()
+                );
+                ex.initCause(e);
```

Comment:
The built exceptions here are not thrown

Reply:
Also thank you!